sig-testing/flakiness-sla.md

```diff
@@ -0,0 +1,91 @@
+# Kubernetes Testing Flakiness SLA
+
+This document captures the expectations of the community about flakiness in
+our tests and our test infrastructure. It sets out an SLA (Service Level 
+Agreement) for flakiness in our tests, as well as actions that we will
+take when we are out of SLA.
+
+
+## Definition of a "Flake"
+
+We'll start by the definition of a _flake_. _Flakiness_ is defined for a
+complete run of one job in the pre-submit testing infrastructure 
+(e.g. pull-kubernetes-e2e-gke-gci) for a given pull request (PR).
+
+A pre-submit job's test result is considered to
+be a flake according to two criteria:
+
+1) it both fails and passes without any changes in the code
+or environment being tested.
+
+2) the PR in question doesn't cause the flake itself.
+
+### Measuring flakiness
+There are a number of challenges in monitoring flakiness. We expect that
+the metric will be heuristic in nature and we will iterate on it over time.
+Identifying all of the potential problems and ways to measure the metric
+are out of the scope of the document, but some currently known challenges
+are listed at the end of the document.
+
+## Flakiness SLA
+We will measure flakiness based on pull requests that are run through pre-submit
+PR testing. The metric that we will monitor is: Pre-submit flakes per PR. This metric
+will be calculated on a daily and weekly basis using:
+
+Sum(Flakes not caused by the PR) / Total(PRs)
+
+Our current SLA is that this metric will be less than 0.01
+(1% of all PRs have flakes that are not caused by the PR itself).
+
+## Activities on SLA violation
+When the project is out of SLA for flakiness on either the daily or weekly metric
+we will stop the merge queue.
+
+Only PRs with the label `queue/fix` or `queue/critical-fix` will be allowed to merge.
+The merge queue will remain blocked until a sufficient number of flake-fixes have been
+submitted to restore the daily CLA. We will measure this each day by via automation that
+re-tests a number of PRs and looking for more flakes.
+
+### Exceptions
+We may consider merging PRs under exceptional circumstances, for example
+merging security fixes or other critical bug fixes that need to be cut into
+a pre-existing release. Each exception will be considered on a case by case
+basis. Exceptions will only be granted for PRs that are already labelled with
+cherry-pick approve.
+
+## Non-goals: Flakiness at HEAD
+We will _not_ currently measure flakiness for continuous e2e that
+are run at HEAD.
+
+There are a number of reasons for this:
+   * We already block the submit-queue if certain HEAD e2e tests are failing
```

Comment:
I want to establish a goal of eliminating this -- we will merge any approved PR that passes PR testing. No CI jobs block the submit queue.